book/entities.rst

```diff
@@ -0,0 +1,1182 @@
+            */
+           public function __construct()
+           {
+           }
+       }
+
+   The class name of an extended entity consists of two parts: Its name always **must** start with
+   ``Extend``. The suffix (here ``Hotel``) must be the name of your entity class.
+
+   The class itself is an empty skeleton. Its actual content will be generated dynamically in the
+   application cache.
+
+#. Let the *entity class* extend the *extend entity* class::
+
+       // src/Acme/DemoBundle/Entity/Hotel.php
+       namespace Acme\DemoBundle\Entity;
+
+       use Acme\DemoBundle\Model\ExtendHotel;
+       use Doctrine\ORM\Mapping as ORM;
+
+       /**
+        * @ORM\Entity
+        * @ORM\Table(name="acme_hotel")
+        */
+       class Hotel extend ExtendHotel
+       {
+           /**
+            * @ORM\Id
+            * @ORM\Column(type="integer")
+            * @ORM\GeneratedValue(strategy="AUTO")
+            */
+           private $id;
+
+           /**
+            * @ORM\Column(type="string", length=255)
+            */
+           private $name;
+
+           public function getId()
+           {
+               return $this->id;
+           }
+
+           public function getName()
+           {
+               return $this->name;
+           }
+
+           public function setName($name)
+           {
+               $this->name = $name;
+           }
+       }
+
+#. Add new properties using Oro migrations::
+
+       // src/Acme/DemoBundle/Migraions/Schema/v2_0;
+       namespace Acme\DemoBundle\Migrations\Schema\v2_0;
+
+       use Doctrine\DBAL\Schema\Schema;
+       use Oro\Bundle\MigrationBundle\Migration\Migration;
+       use Oro\Bundle\MigrationBundle\Migration\QueryBag;
+       use Oro\Bundle\EntityExtendBundle\EntityConfig\ExtendScope;
+
+       class HotelRankingColumn implements Migration
+       {
+           /**
+            * @inheritdoc
+            */
+           public function up(Schema $schema, QueryBag $queries)
+           {
+               $table = $schema->getTable('acme_hotel');
+               $table->addColumn(
+                   'hotel_rating',
+                   'string',
+                   array('oro_options' => array(
+                       'extend' => array(
+                           'is_extend' => true,
+                           'owner' => ExtendScope::OWNER_CUSTOM
+                       ),
+                       'entity' => array('label' => 'Hotel rating'),
+                       'datagrid' => array('is_visible' => false)
+                   ))
+               );
+           }
+       }
+
+   The example above adds a new column ``hotel_ranking``. The third parameter configures the column
+   as an extended field. Using the ``ExtendScope::OWNER_CUSTOM`` owner in the ``oro_options`` key,
+   makes the column visible and configurable in the UI. You can use ``ExtendScope::OWNER_SYSTEM``
+   instead. In that case, it cannot be removed by an administrator.
```

Comment:
I guess that it is important to tell the developer that he/she is fully responsible to adding OWNER_SYSTEM fields to forms, grid and view pages. So, such type of fields provide the developer more control over behaviour of a field, but it can be achieved by writing appropriate code.